### DISCUSSION TITLE
Improve freemods selection UX

### DISCUSSION THREAD

**Discussion by Gonzalo-Bruna** [ppy/osu#24245] [Ideas]
After playing a mutiplayer lobby, I realized I didn't even know how to pick freemods. Even after been playing osu! for the past 11 years, I couldn't tell in lazer UI. The interface is not intuitive, if it was hard for me I can't think of a guy that just started playing the game.

At first, I thought it was under the lobby settings, but then I realized you have to enter the song select screen and select the ones you want. Not only that, but if you select all free mods and then press the back button, none of the freemods will remain selected, you have to press "enter" to enable the freemods, and that means there is a chance you might change the currently selected beatmap by mistake.

But that's not all.  Freemods are enabled per-beatmap, making it even more confusing, cause you might think you already selected freemods but you actually did for just one beatmap.

My idea is to put it as a dropdown in the lobby settings, allowing to select freemods "per-beatmap" or "global". For the "global" option, the default should be all freemods selected, and if people wants to customize it further, then a "customize" button could be added that would lead to the song select screen and opens the freemods tab.

This is just and idea, I'm not an expert but you guys might agree with some of these changes.

**Comment by bdach** [comment]
> Not only that, but if you select all free mods and then press the back button, none of the freemods will remain selected, you have to press "enter" to enable the freemods

Okay, the fact that you have to essentially re-select the beatmap is not great, I agree.

> But that's not all. Freemods are enabled per-beatmap, making it even more confusing, cause you might think you already selected freemods but you actually did for just one beatmap.

This is functionality, though. Maybe it needs to be shown better but I wouldn't make two "types" of free mods.

**Reply by Gonzalo-Bruna** [reply]
I mean, it's just an idea, probably you guys know better what a good solution is. What I wanted to point out is the bad UX of selecting freemods, it's not intuitive at all for most players.

**Comment by peppy** [comment]
I agree with having “room defaults” for mods

**Comment by delan** [comment]
Yeah, the mods and freemods ui makes sense to me now, but I had to google it. Even then, I thought freemods were broken for me, until I figured out the “extra mods” section only shows when the song with freemods reaches the top of the queue.

The back button thing also tripped up a friend of mine who’s newer to osu. They wanted to keep the same song, but toggle freemods, so they clicked edit,

![image](https://github.com/ppy/osu/assets/465303/c8a98f71-3eb7-49d5-8ebd-cb656d2b9fda)

then toggled freemods from off to all,

![image](https://github.com/ppy/osu/assets/465303/2dcd55ee-07c9-4852-9e15-4a2b869d1cb4)

then clicked back,

![image](https://github.com/ppy/osu/assets/465303/acd15cf7-ee66-4f36-a8eb-2ed3af27429c)

but because they clicked back instead of rechoosing the song, the freemods didn’t stick. It took us a few tries to figure out what was happening, so I wonder if we want some kinda unsaved changes warning for that?

**Reply by peppy** [reply]
The back button should probably save changes, yeah.

**Comment by marioman63** [comment]
> But that's not all. Freemods are enabled per-beatmap, making it even more confusing, cause you might think you already selected freemods but you actually did for just one beatmap.

i agree the interface confused me at first as a 10+ year stable player, but once you set the mods you want on your song select screen, ive never had to set them again unless I leave the lobby. if i enter song select again, my free mods are still selected

**Comment by Gonzalo-Bruna** [comment]
it probably would be nice to have an option as a toggle in the lobby settings so we can choose to select free mods per-beatmap or general (with general I mean just like stable).
